### PR TITLE
e2e: Upgrade snapshot clientset and api from v1beta1 to v1

### PR DESCRIPTION
**Pull request by humblec**
snapshot.go currently make use of snapshot v1beta1 clientset and api, it has been rolled into v1 version of clientset and api .

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,8 @@ require (
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/kubernetes-csi/csi-lib-utils v0.9.1
-	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1
+	github.com/kubernetes-csi/external-snapshotter/client/v4 v4.1.0
+	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.5
```
Comment (Madhu-1) [inline go.mod]: Is this needed anymore?

**Comment by Madhu-1** [comment]
golangci-lint failure

File: e2e/snapshot.go [hunk 2]
```diff
@@ -6,8 +6,8 @@ import (
 	"strings"
 	"time"
 
-	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
-	snapclient "github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned"
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v4/clientset/versioned/typed/volumesnapshot/v1"
```
Comment (Madhu-1) [inline e2e/snapshot.go]: Any specific reason to change the alias/ if not can we keep this as same.

File: go.mod [hunk 3]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/kubernetes-csi/csi-lib-utils v0.9.1
-	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1
+	github.com/kubernetes-csi/external-snapshotter/client/v4 v4.1.0
```
Comment (Madhu-1) [inline go.mod:17]: as we have 4.1.1 https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v4.1.1 can  we update this to 4.1.1?

File: go.mod [hunk 4]
```diff
@@ -39,7 +39,8 @@ replace (
 	github.com/golang/protobuf => github.com/golang/protobuf v1.4.3
 	github.com/hashicorp/vault/api => github.com/hashicorp/vault/api v1.0.5-0.20200902155336-f9d5ce5a171a
 	github.com/hashicorp/vault/sdk => github.com/hashicorp/vault/sdk v0.1.14-0.20201116234512-b4d4137dfe8b
-	github.com/kubernetes-csi/external-snapshotter/v2 => github.com/kubernetes-csi/external-snapshotter/v2 v2.1.1-0.20200504125226-859696c419ff
+	github.com/kubernetes-incubator/external-storage => github.com/kubernetes-incubator/external-storage v5.5.0+incompatible
+	google.golang.org/grpc => google.golang.org/grpc v1.35.0
```
Comment (Madhu-1) [inline go.mod]: Any specific reason to add this here? this got removed in https://github.com/ceph/ceph-csi/pull/2179/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L43

File: e2e/snapshot.go (hunk 2 above)
Comment (humblec) [reply e2e/snapshot.go]: sure.. reverted.

File: e2e/snapshot.go [hunk 5]
```diff
@@ -6,8 +6,8 @@ import (
 	"strings"
 	"time"
 
-	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
-	snapclient "github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned"
+	snapclient "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
```
Comment (nixpanic) [inline e2e/snapshot.go]: you changed other places to `snapapi`, better to use the same alias everywhere?

File: e2e/snapshot.go (hunk 2 above)
Comment (nixpanic) [reply e2e/snapshot.go]: why not stick with `snapclient`?

**Review by nixpanic — DISMISSED** [review]

File: e2e/snapshot.go (hunk 5 above)
Comment (humblec) [reply e2e/snapshot.go]: @nixpanic I dont get above comment, in e2e files , its `snapapi` everywhere. did I miss any?

File: e2e/snapshot.go (hunk 2 above)
Comment (humblec) [reply e2e/snapshot.go]: @nixpanic its `snapclient` version, where do you see different import name?

File: go.mod (hunk 3 above)
Comment (humblec) [reply go.mod:17]: @Madhu-1 eventhough the tag and release is available, the client package resolution to `4.1.1` was not happening to me. Is it resolving/importing `4.1.1` properly at your end ?

Comment (Madhu-1) [reply go.mod:17]: what is the error. Maybe we can check with the kube-csi team?

File: e2e/snapshot.go (hunk 5 above)
Comment (nixpanic) [reply e2e/snapshot.go]: You updated to use
```
snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
```
 everywhere else. Here you rename the alias `snapapi` to `snapclient`. Better use `snapapi` here as it used to be, and what you used in other locations.

File: go.mod (hunk 3 above)
Comment (humblec) [reply go.mod:17]: here it is : ```go: github.com/kubernetes-csi/external-snapshotter/client/v4@v4.1.1: reading github.com/kubernetes-csi/external-snapshotter/client/client/go.mod at revision client/v4.1.1: unknown revision client/v4.1.1``` , I can check with csi team though

File: e2e/snapshot.go (hunk 5 above)
Comment (humblec) [reply e2e/snapshot.go]: ok. will have the same notion here too.. changed in the last refresh

File: go.mod (hunk 3 above)
Comment (nixpanic) [reply go.mod:17]: I agree that it is a little strange, and ideally we should use v4.1.1.

That said there is only kubernetes-csi/external-snapshotter#515 addressed, which does not seem to affect the client package.

**Review by nixpanic — APPROVED** [review]

**Comment by humblec** [comment]
@Madhu-1 can you please review ?

**Review by Madhu-1 — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio rebase

**Pull request merged**